baybe/campaign.py

```diff
@@ -528,11 +533,76 @@ def posterior(self, candidates: pd.DataFrame) -> Posterior:
                 f"provide a '{method_name}' method."
             )
 
-        import torch
-
         with torch.no_grad():
             return surrogate.posterior(candidates)
 
+    def posterior_stats(
+        self, candidates: pd.DataFrame, stats: Sequence[Statistic] | None = None
+    ) -> pd.DataFrame:
+        """Return common posterior statistics for each target.
+
+        Args:
+            candidates: The candidate points in experimental representation.
+                For details, see :meth:`baybe.surrogates.base.Surrogate.posterior`.
+            stats: Sequence indicating which statistics to compute. Also accepts
+                floats, for which the corresponding quantile point will be computed.
+
+        Raises:
+            ValueError: If a requested quantile is outside the open interval (0,1).
+            TypeError: If the posterior utilized by the surrogate does not support
+                a requested statistic.
+
+        Returns:
+            Data frame with prediction statistics for each target for each candidate.
+        """
+        stats = stats or ["mean", "std"]
+        for stat in (x for x in stats if isinstance(x, float)):
+            if not 0 < stat < 1.0:
+                raise ValueError(
+                    f"Posterior quantile statistics can only be computed for quantiles "
+                    f"between 0 and 1 (non-inclusive). Provided value: '{stat}' as "
+                    f"part of {stats=}'."
+                )
+        posterior = self.posterior(candidates)
+
+        assert self.objective is not None
+        match self.objective:
+            case DesirabilityObjective():
+                # TODO: Once desirability also supports posterior transforms this check
+                #  here will have to depend on the configuration of the objective and
+                #  whether it uses the transforms or not.
+                targets = ["Desirability"]
+            case _:
+                targets = [t.name for t in self.objective.targets]
+
+        result = pd.DataFrame(index=candidates.index)
+        for k, t in enumerate(targets):
```

Comment:
let's use explicit variable names
```suggestion
        for idx, target_name in enumerate(targets):
```

Reply:
I accepted the target change but d rather not name a running index idx (id only use that for dataframe related index things)